packages/analytics-core/src/pluginTypeDef.js

```diff
@@ -1,12 +1,71 @@
+/**
+ * @callback PluginTrackFunction
+ * @param {Object} arg
+ * @param {Object} arg.config config from the plugin spec
+ * @param {AnalyticsInstance} arg.instance analytics instance
+ * @param {Object} arg.payload event data
+ * @param {string} arg.payload.event event name passed to track
+ * @param {TrackEventProperties} [arg.payload.properties] event properties passed to track
```

Comment:
Not sure if this is actually optional.  Will the library always supply an object here?